src/events/publisher.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <functional>
+#include <string>
+
+#include "basic_event.h"
+
+namespace events {
+class Publisher {
+public:
+	template <typename T> using event_type = basic_event<T>;
+
+	using transport_type = std::function<void(const std::string &)>;
+
+	Publisher()                  = default;
+	Publisher(const Publisher &) = delete;
```

Comment:
what does this do?

Reply:
The `default` is to ask the compiler to define the implicit default constructor (so it can initialise `_transport` member variable). 

More importantly `Publisher(const Publisher &) = delete;` is to not allow copying the `Publisher` so `events::publisher()` returns a singleton and `events::publish()` reuses the same instance.

Ref: https://en.cppreference.com/w/cpp/language/default_constructor